pkg/volume/cinder/attacher.go

```diff
@@ -68,9 +69,13 @@ func (attacher *cinderDiskAttacher) Attach(spec *volume.Spec, nodeName types.Nod
 
 	volumeID := volumeSource.VolumeID
 
-	instanceid, err := attacher.nodeInstanceID(nodeName)
+	instanceid, err := getNodeInstanceID(attacher.host, nodeName)
```

Comment:
It's a pity this Attach() function isn't just called with a full node object rather than just the name, since I suspect some caller somewhere already has that full object available...   What would it take to modify the caller(s) to pass down the full node object instead?